### PR TITLE
1.0.0 branch PR

### PR DESCRIPTION
**Pull request by hartzis**
PR to track `1.x` branch progress for `1.0.0` release and will close issue #91 when merged. 🎉 

- [x] use 1.0.0 react-async-script
- [x] update readme
- [x] update tests
- [x] fill out "skipped" tests for `Expired` functionality
- [x] add support for the "error-callback" - #93
  - [x] add tests for error-callback
- [x] only support LTS or soon to be LTS (10) for travis ci
  - https://github.com/nodejs/Release
  - only node 8, 10
- [x] update test runner to `jest` - #95 
  - i think we may be able to do this very easily
  - maybe add `enzyme` if needed
- [x] update lint and add prettier
- [x] update `removeOnMount` to `removeOnUnmount` - [react-async-script/api](https://github.com/dozoisch/react-async-script#async-script-hoc-api) 😸

### CODE REVIEW

**Comment by hartzis** [comment]
@dozoisch if you're able to cut a release candidate from the `1.x` branch thad be sweet 😺

**Comment by dozoisch** [comment]
Perfect! Will cut a @next release

**Comment by dozoisch** [comment]
@hartzis alright rc.2 should be good =) rc.1 doesn't have the correct build files

**Comment by hartzis** [comment]
@dozoisch thank you, you rock! It looks like we need to add the same shims we did in `react-async-script` for the tests, lol. Looks like we also may need to update `eslint` since it doesn't know how to handle a scoped `class` correctly 🤣 

Also want to update the node versions for travis to match `react-async-script`

**Comment by hartzis** [comment]
just a heads up - Working `1.0.0` release candidate - https://codesandbox.io/s/1y4zzjq37l

**Comment by hartzis** [comment]
well... the tests and lint pass locally. travis seems to be having trouble with `node` 8/10 with `yarn` and `phantomJS`.

This might be the opportunity to update to `enzyme`/`jest`? Shouldn't be too bad. may even be able to just do jest and no enzyme and keep the test exactly like they are?

**Comment by dozoisch** [comment]
Jest is fine with me ! The stacked used here is pretty old 😅😅 so I’m all for modernizing it!

File: package.json [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
+  "main": "lib/index.js",
+  "module": "lib/es/index.js",
   "directories": {
     "lib": "lib/"
   },
   "scripts": {
     "build": "rm -rf lib && npm run build:cjs && npm run build:es",
     "build:cjs": "babel src --out-dir lib",
     "build:es": "BABEL_ENV=es babel src --out-dir lib/es",
+    "prepublish": "npm run build",
```
[the remark this answers is not in the document]
Comment (hartzis) [reply package.json]: example of npm `prepare`

https://github.com/reduxjs/react-redux/blob/47facdbcae2412d37d72fa5c37cb4e0cd64a9b4a/package.json#L37

[the remark this answers is not in the document]
Comment (hartzis) [reply package.json]: apologies about the notification spam if it happened, was trying to get the comment code snippet to work, but i don't think it works cross repo...

**Comment by dozoisch** [comment]
Alright eslint/prettier is done! I'll go do the same in react-async-script

**Comment by hartzis** [comment]
cut another `rc`? 😸

**Comment by dozoisch** [comment]
Will do!

**Comment by dozoisch** [comment]
Done!

**Comment by hartzis** [comment]
@dozoisch `rc.3` looking good - https://codesandbox.io/s/1y4zzjq37l

**Comment by hartzis** [comment]
last thing is to release `1.0.0` of `react-async-script` before we cut official `1.0.0` of this

**Comment by dozoisch** [comment]
@hartzis 🙌 published 1.0.0 on both repos :)! Awesome work

**Pull request merged**